okhttp/src/main/java/okhttp3/NewWebSocket.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3;
+
+import okio.ByteString;
+
+/**
+ * A non-blocking interface to a websocket. Use the {@link NewWebSocket.Factory factory} to create
+ * instances; usually this is {@link OkHttpClient}.
+ *
+ * <h3>Websocket Lifecycle</h3>
+ *
+ * Upon normal operation each websocket progresses through a sequence of states:
+ *
+ * <ul>
+ *   <li><strong>Connecting:</strong> the initial state of each websocket. Messages may be enqueued
+ *       but they won't be transmitted until the websocket is open.
+ *   <li><strong>Open:</strong> the websocket has been accepted by the remote peer and is fully
+ *       operational. Messages in either direction are enqueued for immediate transmission.
+ *   <li><strong>Closing:</strong> one of the peers on the websocket has initiated a graceful
+ *       shutdown. The websocket will continue to transmit already-enqueued messages but will refuse
```

Comment:
Again, I'm not sure if "closing" includes the client calling `close()`. But if it does, then the client probably doesn't want it to keep transmitting already-enqueued messages.